AutoGraphTests/WebSocketClientTests.swift

```diff
@@ -0,0 +1,349 @@
+import XCTest
+@testable import Starscream
+@testable import AutoGraphQL
+import JSONValueRX
+
+private let kDelay = 0.5
+
+class WebSocketClientTests: XCTestCase {
+    var subject: MockWebSocketClient!
+    var webSocket: MockWebSocket!
+    var webSocketDelegate: MockWebSocketClientDelegate!
+    
+    override func setUp() {
+        super.setUp()
+        let url = URL(string: "localhost")!
+        self.webSocket = MockWebSocket(request: URLRequest(url: url))
+        self.subject = try! MockWebSocketClient(url: url, webSocket: self.webSocket)
+        self.webSocketDelegate = MockWebSocketClientDelegate()
+        self.subject.delegate = self.webSocketDelegate
+    }
+    
+    override func tearDown() {
+        self.subject = nil
+        self.webSocket = nil
+        self.webSocketDelegate = nil
+        super.tearDown()
+    }
+    
+    func testSendSubscriptionResponseHandlerIsCalledOnSuccess() {
+        var called = false
+        let request = try! SubscriptionRequest(request: FilmSubscriptionRequest(), operationName: "film")
+        _ = self.subject.subscribe(request: request, responseHandler: SubscriptionResponseHandler(completion: { result in
+            
+            guard case .success(_) = result else {
+                XCTFail()
+                return
+            }
+            
+            called = true
+        }))
+        
+        waitFor(delay: kDelay)
+        XCTAssertTrue(called)
+    }
+    
+    
+    func testSendSubscriptionCorrectlyDecodesResponse() {
+        var film: Film?
+        let request = try! SubscriptionRequest(request: FilmSubscriptionRequest(), operationName: "film")
+        _ = self.subject.subscribe(request: request, responseHandler: SubscriptionResponseHandler(completion: { result in
+            
+            guard case let .success(data) = result else {
+                XCTFail()
+                return
+            }
+            
+            film = try! JSONDecoder().decode(Film.self, from: data)
```

Comment:
we should decode using the websocket's serializer so that get's tested here too. then do an equality assertion over the serialized object we get back